### PR TITLE
laravel fix and auth working

### LINKED ISSUE #5: Authentication
Make controller and endpoint for login, register and log out .

### LINKED ISSUE #27: Host API
Host API

### PR DESCRIPTION
**Pull request by AndrzejBudzyn**
Some fixes to laravel and working auth for now the endpoints are

/login [POST] email,password
/register [POST] name,email,password  password_confirmation
/logout [POST]
/getCurrentUser [GET]

And our api route is 
https://sponge-climbing-adder.ngrok-free.app/api

### CODE REVIEW

File: Backend/Dockerfile
Comment (PiotrFedak) [inline Backend/Dockerfile:1]: Do we need this ?

File: Backend/composer.json [hunk 1]
```diff
@@ -8,18 +8,22 @@
   ],
   "license": "MIT",
   "require": {
-    "php": "^8.2",
-    "laravel/framework": "^11.9",
-    "laravel/tinker": "^2.9"
+    "php": "^8.1",
+    "guzzlehttp/guzzle": "^7.2",
+    "laravel/framework": "^10.10",
```
Comment (PiotrFedak) [inline Backend/composer.json:13]: why did you downgrade laravel versions ?

File: Backend/config/services.php [hunk 2]
```diff
@@ -15,24 +15,32 @@
...
+        "redirect" => "http://127.0.0.1:8000/auth/github/callback",
     ],
 
-    "resend" => [
-        "key" => env("RESEND_KEY"),
+    "google" => [
+        "client_id" => env("GOOGLE_CLIENT_ID"),
+        "client_secret" => env("GOOGLE_CLIENT_SECRET"),
+        "redirect" => "http://127.0.0.1:8000/auth/google/callback",
     ],
```
Comment (PiotrFedak) [inline Backend/config/services.php]: delete it

File: Backend/config/services.php [hunk 3]
```diff
@@ -15,24 +15,32 @@
...
 
-    "ses" => [
-        "key" => env("AWS_ACCESS_KEY_ID"),
-        "secret" => env("AWS_SECRET_ACCESS_KEY"),
-        "region" => env("AWS_DEFAULT_REGION", "us-east-1"),
+    "github" => [
+        "client_id" => env("GITHUB_CLIENT_ID"),
+        "client_secret" => env("GITHUB_CLIENT_SECRET"),
+        "redirect" => "http://127.0.0.1:8000/auth/github/callback",
```
Comment (PiotrFedak) [inline Backend/config/services.php]: are you planning to create a sign up with github?

File: Backend/config/services.php [hunk 4]
```diff
@@ -15,24 +15,32 @@
     |
     */
 
+    "mailgun" => [
+        "domain" => env("MAILGUN_DOMAIN"),
+        "secret" => env("MAILGUN_SECRET"),
+        "endpoint" => env("MAILGUN_ENDPOINT", "api.mailgun.net"),
+        "scheme" => "https",
+    ],
+
```
Comment (PiotrFedak) [inline Backend/config/services.php]: 👀

**Review by PiotrFedak — DISMISSED** [review]
Please take a moment to clean unnecessary files.
Apart from that, I tested this PR and the endpoints works fine

File: Backend/composer.json (hunk 1 above)
Comment (AndrzejBudzyn) [reply Backend/composer.json:13]: for better work for me

**Comment by KacperWojdak** [comment]
Git

**Review by KacperWojdak — APPROVED** [review]
Git

**Pull request merged**